### ISSUE TITLE
docs not showing correct version

### ISSUE THREAD

**Issue by trichter** [obspy/obspy#2956]
@ThomasLecocq 

The doc version you uploaded at msnoise.org is showing the correct version number, but this appears not to be the case for the doc configuration in master. E.g. https://github.com/obspy/obspy/actions/runs/1714061599

![obs_missing_version](https://user-images.githubusercontent.com/2046150/150008808-175ac387-f40f-4130-a3a6-c1fbec9dc1f7.png)

**Comment by ThomasLecocq** [comment]
It's because `master` has no "version"

https://github.com/obspy/obspy/blob/8c838cc6a56cecea65542bd5236bf4337f88eea8/obspy/core/util/version.py#L153-L155

**Comment by trichter** [comment]
Maybe it's because git is not installed in the docs container. Could we try to switch to the conda/mamba installation process? It's not so much overhead compared to the docs build time.

**Comment by ThomasLecocq** [comment]
agree !

**Comment by trichter** [comment]
Alos, test reports do not show the correct version string. It started failing around the time when we switch from travis to github.

```
#112264 | 1 | 0.0.0.dev+0.g45206c13c1 | #2740 | 1943 | 49 | gh-actions | 3.8.6 | Linux (64bit) | a year ago
#112263 | 1 | 0.0.0.dev+0.g6bb7c1bf35 | #2740 | 1943 | 49 | gh-actions | 3.7.8 | Linux (64bit) | a year ago
#112262 | 30 | 1.2.2.post0+32.g2f990f92c2.obspy.m… | - | 2188 | 58 | travis-ci | 3.8.6 | Darwin (64bit) | a year ago
#112261 | 29 | 1.2.2.post0+32.g2f990f92c2.obspy.m… | - | 2188 | 58 | travis-ci | 3.7.8 | Darwin (64bit) | a year ago
```

Locally, its working for me.

```
In [2]: obspy.__version__
Out[2]: '1.2.2.post0+538.g7039b83550'
```

**Comment by trichter** [comment]
I think we can fix the first (docs) issue by installing git.

The second is probably an incarnation of https://github.com/obspy/obspy/commit/b437f312883bc1e4f8b0b2697c06e3604ee45f0c#commitcomment-2278698 mentioned in #482.

We probably just need to set `fetch-depth: 0` in the checkout actions.

**Comment by d-chambers** [comment]
closed by #2969
